src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,99 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                       PreparedStatement preparedStatement = connection.prepareStatement(query,
+                        PreparedStatement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                book.setId(resultSet.getLong(1));
```

Comment:
```suggestion
                Long id = generatedKeys.getObject(1, Long.class);
                book.setId(id);
```

Check common mistakes file